pkg/controller/gc_controller.go

```diff
@@ -159,53 +130,56 @@ func (c *gcController) processQueueItem(key string) error {
 		} else {
 			backup.Labels[garbageCollectionFailure] = gcFailureBSLCannotGet
 		}
-		if err := c.kbClient.Update(context.Background(), backup); err != nil {
+		if err := c.Update(ctx, backup); err != nil {
 			log.WithError(err).Error("error updating backup labels")
 		}
-		return errors.Wrap(err, "error getting backup storage location")
+		return ctrl.Result{}, errors.Wrap(err, "error getting backup storage location")
 	}
 
 	if loc.Spec.AccessMode == velerov1api.BackupStorageLocationAccessModeReadOnly {
-		log.Infof("Backup cannot be garbage-collected because backup storage location %s is currently in read-only mode", loc.Name)
+		log.Infof("gBackup cannot be garbage-collected because backup storage location %s is currently in read-only mode", loc.Name)
 		backup.Labels[garbageCollectionFailure] = gcFailureBSLReadOnly
-		if err := c.kbClient.Update(context.Background(), backup); err != nil {
+		if err := c.Update(ctx, backup); err != nil {
 			log.WithError(err).Error("error updating backup labels")
 		}
-		return nil
+		return ctrl.Result{}, nil
 	}
 
 	// remove gc fail error label after this point
 	delete(backup.Labels, garbageCollectionFailure)
-	if err := c.kbClient.Update(context.Background(), backup); err != nil {
+	if err := c.Update(ctx, backup); err != nil {
 		log.WithError(err).Error("error updating backup labels")
 	}
 
-	selector := labels.SelectorFromSet(labels.Set(map[string]string{
+	selector := client.MatchingLabels{
 		velerov1api.BackupNameLabel: label.GetValidName(backup.Name),
 		velerov1api.BackupUIDLabel:  string(backup.UID),
-	}))
+	}
 
-	dbrs, err := c.deleteBackupRequestLister.DeleteBackupRequests(ns).List(selector)
-	if err != nil {
-		return errors.Wrap(err, "error listing existing DeleteBackupRequests for backup")
+	dbrs := &velerov1api.DeleteBackupRequestList{}
+	if err := c.List(ctx, dbrs, selector); err != nil {
+		log.WithError(err).Error("error listing DeleteBackupRequests")
+		return ctrl.Result{}, errors.Wrap(err, "error listing existing DeleteBackupRequests for backup")
 	}
+	log.Debugf("dbrs:%v", dbrs)
```

Comment:
This may be a long list, although this is a debug level log.
Suggest to print the length instead.

Reply:
Sure, thanks.